lib/immediate-juttle-job.js

```diff
@@ -43,6 +46,8 @@ class ImmediateJuttleJob extends JuttleJob {
             // We actually wait until close() to return the data.
         } else if (msg.type === 'error') {
             self._errors.push(msg.error);
+        } else if (msg.type === 'log') {
+            self._logs.push(msg);
```

Comment:
You could save some space by checking debug here and only saving it when debug is true, rather than always saving it and including it in the result when debug is true.